src/Microsoft.VisualStudio.Services.Agent/SecretMasker.cs

```diff
@@ -2,30 +2,29 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
+using System.Text.RegularExpressions;
 
 namespace Microsoft.VisualStudio.Services.Agent
 {
     [ServiceLocator(Default = typeof(SecretMasker))]
     public interface ISecretMasker : IAgentService
     {
-        IEnumerable<ISecret> Masks { get; }
-
         string MaskSecrets(string input);
 
-        void Add(ISecret mask);
-
-        void AddRegEx(string expression);
+        void AddRegex(string expression);
 
         void AddValue(string value);
 
-        void AddVariableName(string variableName, string value);
+        void AddVariable(string name, string value);
     }
 
-    public class SecretMasker : AgentService, ISecretMasker
+    public sealed class SecretMasker : AgentService, ISecretMasker
     {
         private const string Mask = "********";
+        private readonly List<ISecret> _secrets = new List<ISecret>();
+        private readonly HashSet<string> _variableNames = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
 
-        private readonly List<ISecret> _secretMasks = new List<ISecret>();
+        public HashSet<string> VariableNames => _variableNames;
```

Comment:
This will be required when adding secrets from the command handler. I put a good deal of thought into how it should work, but did not fully integrate the code.